### DISCUSSION TITLE
Getting Solution Items Guid

### DISCUSSION THREAD

**Discussion by scottcantwell** [VsixCommunity/Community.VisualStudio.Toolkit#390] [Q&A] [answered]
Is there a way to get the GUID for the Solution Items folder?

**Comment by reduckted** [comment, marked as answer]
As far as I'm aware, the "Solution Items" folder is nothing more than a solution folder with the name "Solution Items".

The GUID for a solution folder is defined in `Microsoft.VisualStudio.VSConstants.CLSID.SolutionFolderProject_guid`

**Reply by scottcantwell** [reply]
I am looking for the GUID on the right, not the Project Type GUID.

Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution Items", "{1E9AA769-E1CD-4FB4-B07A-2E043FA8135B}"

**Reply by reduckted** [reply]
Ah, sorry, I misunderstood what you meant. 

The GUID of a project (a solution folder is just a type of project) can be retrieved using the `IVsSolution.GetGuidOfProject` method. TO call that, you need the `IVsHierarchy` of the project.

There's a few ways that you can do this, but this is probably the simplest way:

```cs
private async Task<Guid?> GetSolutionFolderGuidAsync()
{
    await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();

    // Loop through the children of the solution and find the `SolutionFolder` that has the name "Solution Items".
    SolutionItem folder = (await VS.Solutions.GetCurrentSolutionAsync())
        .Children
        .Where((x) => x.Type == SolutionItemType.SolutionFolder)
        .Where((x) => string.Equals(x.Name, "Solution Items"))
        .FirstOrDefault();

    if (folder is not null)
    {
        // Get the IVsHierarchy for that solution folder, then get the Guid for that hierarchy.
        folder.GetItemInfo(out IVsHierarchy hierarchy, out _, out _);

        if (ErrorHandler.Succeeded((await VS.Services.GetSolutionAsync()).GetGuidOfProject(hierarchy, out Guid guid)) {
            return guid;
        }
    }

    return null;
}
```

**Reply by scottcantwell** [reply]
"Ah, sorry, I misunderstood what you meant."

I should have been clearer in my question.  Anyways, your solution worked. Thanks!

**Comment by RobertvanderHulst** [comment]
The first guid is that of a solution folder ({2150E333-8FDC-42A3-9474-1A3956D46DE8}) the other guid is unique per solution.
